eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/FeeEstimator.scala

```diff
@@ -25,10 +28,51 @@ trait FeeEstimator {
   // @formatter:on
 }
 
+object FeeEstimator {
+  /** When using anchor outputs, we only need to set a feerate that allows the tx to propagate: we will use CPFP to speed up confirmation if needed. */
+  val AnchorOutputMaxCommitFeerate = FeeratePerKw(FeeratePerByte(10 sat))
+}
+
 case class FeeTargets(fundingBlockTarget: Int, commitmentBlockTarget: Int, mutualCloseBlockTarget: Int, claimMainBlockTarget: Int)
 
-case class FeerateTolerance(ratioLow: Double, ratioHigh: Double)
+case class FeerateTolerance(ratioLow: Double, ratioHigh: Double) {
+  /**
+   * @param channelVersion  channel version
+   * @param networkFeerate  reference fee rate (value we estimate from our view of the network)
+   * @param proposedFeerate fee rate proposed (new proposal through update_fee or previous proposal used in our current commit tx)
+   * @return true if the difference between proposed and reference fee rates is too high.
+   */
+  def isFeeDiffTooHigh(channelVersion: ChannelVersion, networkFeerate: FeeratePerKw, proposedFeerate: FeeratePerKw): Boolean = {
+    if (channelVersion.hasAnchorOutputs) {
+      proposedFeerate < networkFeerate * ratioLow || FeeEstimator.AnchorOutputMaxCommitFeerate * ratioHigh < proposedFeerate
```

Comment:
Shouldn't the `AnchorOutputMaxCommitFeerate` be an hard limit? Or do you consider that `UpdateFee` contains the actual network feerate, and we will cap it with `AnchorOutputMaxCommitFeerate`? The problem I have with the latter is that there is no such thing as a reasonable network feerate, since it all depends on CPFP.

```suggestion
      proposedFeerate < networkFeerate * ratioLow || FeeEstimator.AnchorOutputMaxCommitFeerate < proposedFeerate
```

Reply:
The problem is that we don't control what value our remote will use there (when they're funder).
LND uses 10 sat/byte by default, but they may change it to 5 sat/byte or 20 sat/byte.
If we want to use 10 sat/byte for the channels where we're funder and allow some leeway for the channels where we're fundee, we need to use a ratio here.

The only other option I see is to blindly accept the remote feerate when we're fundee, do you think that's better (it's a reasonable alternative in my opinion)? I think we can only blindly accept high feerates, we still want to ensure our remote doesn't use a feerate that's so low that the tx will not even relay...

Reply:
> The problem is that we don't control what value our remote will use there (when they're funder).
LND uses 10 sat/byte by default, but they may change it to 5 sat/byte or 20 sat/byte.
If we want to use 10 sat/byte for the channels where we're funder and allow some leeway for the channels where we're fundee, we need to use a ratio here.

Nevermind, I thought the `AnchorOutputMaxCommitFeerate` value was defined in spec. With that in mind, we either set a tolerance, or remove the check alltogether as you propose below.

> The only other option I see is to blindly accept the remote feerate when we're fundee, do you think that's better (it's a reasonable alternative in my opinion)? I think we can only blindly accept high feerates, we still want to ensure our remote doesn't use a feerate that's so low that the tx will not even relay (at least until we get package relay)...but even that may not be correctly enforceable, or we may be too late when we realize that the fee isn't enough :/

True, since the funder pays, I don't see why we wouldn't let him pay as much as he wants either. Not sure why we need this notion of a higher limit on the fees.

Reply:
> Not sure why we need this notion of a higher limit on the fees.

I think if we use a tolerance, it still makes sense to have a higher bound as well, because by raising the fee the remote peer can make all HTLC outputs become dust / non-enforceable on-chain, which can be undesirable.

Reply:
Let's stick with this tolerance ratio for now and re-visit later once we're doing interop tests with other implementations and see how they behave?

Reply:
Fair enough